bioconductor-alabaster.base: fix osx-64 build

aliciaaevans opened the pull request:
Issue seems to be related to OSX SDK because the error complained about a feature only in 10.15. However, we have the environment variable set to 10.15.

VinzentRisch commented:
Hey @aliciaaevans 
I would need the osx64 build of this package for another package. Is there any plan to fix this is the near future? Would you be willing to give it another go, or who would be a good person to try to contact with this issue? I would love to help but my experience with this kind of problem is near zero and I don't think I can manage it alone. 
Thank You! :)

aliciaaevans commented:
Converted to draft while I try some stuff that shouldn't be merged

aliciaaevans commented:
I've tried everything I can think of to force the 10.15 SDK, but it's still getting ` -mmacosx-version-min=10.13` from somewhere. I know we have other recipes that require 10.15, but none of those strategies seem to be helping. I may continue flailing around on this, but if anyone else wants to take a stab at it, please feel free.

```
20:10:45 BIOCONDA INFO (ERR) ../inst/include/ritsuko/hdf5/open.hpp:31:55: error: 'path' is unavailable: introduced in macOS 10.15 - see https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
20:10:45 BIOCONDA INFO (ERR)    31 |     if constexpr(std::is_same<Path_, std::filesystem::path>::value) {
20:10:45 BIOCONDA INFO (ERR)       |                                                       ^
20:10:45 BIOCONDA INFO (ERR) /opt/mambaforge/envs/bioconda/conda-bld/bioconductor-alabaster.base_1752264365769/_build_env/bin/../include/c++/v1/__filesystem/path.h:383:33: note: 'path' has been explicitly marked unavailable here
20:10:45 BIOCONDA INFO (ERR)   383 | class _LIBCPP_EXPORTED_FROM_ABI path {
20:10:45 BIOCONDA INFO (ERR)       |                                 ^
20:10:45 BIOCONDA INFO (ERR) In file included from check_list.cpp:3:
20:10:45 BIOCONDA INFO (ERR) In file included from ../inst/include/uzuki2/uzuki2.hpp:15:
20:10:45 BIOCONDA INFO (ERR) In file included from ../inst/include/uzuki2/parse_hdf5.hpp:22:
20:10:45 BIOCONDA INFO (ERR) In file included from ../inst/include/ritsuko/hdf5/hdf5.hpp:15:
20:10:45 BIOCONDA INFO (ERR) In file included from ../inst/include/ritsuko/hdf5/miscellaneous.hpp:7:
20:10:45 BIOCONDA INFO (ERR) ../inst/include/ritsuko/hdf5/open.hpp:44:55: error: 'path' is unavailable: introduced in macOS 10.15 - see https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
20:10:45 BIOCONDA INFO (ERR)    44 |     if constexpr(std::is_same<Path_, std::filesystem::path>::value) {
20:10:45 BIOCONDA INFO (ERR)       |                                                       ^
20:10:45 BIOCONDA INFO (ERR) /opt/mambaforge/envs/bioconda/conda-bld/bioconductor-alabaster.base_1752264365769/_build_env/bin/../include/c++/v1/__filesystem/path.h:383:33: note: 'path' has been explicitly marked unavailable here
20:10:45 BIOCONDA INFO (ERR)   383 | class _LIBCPP_EXPORTED_FROM_ABI path {
20:10:45 BIOCONDA INFO (ERR)       |                                 ^
20:10:46 BIOCONDA INFO (ERR) 2 errors generated.
20:10:46 BIOCONDA INFO (ERR) make: *** [/opt/mambaforge/envs/bioconda/conda-bld/bioconductor-alabaster.base_1752264365769/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/R/etc/Makeconf:204: check_list.o] Error 1

...

20:10:46 BIOCONDA INFO (OUT) x86_64-apple-darwin13.4.0-clang++ -std=gnu++17 -I"$PREFIX/lib/R/include" -DNDEBUG -I../inst/include -I'$PREFIX/lib/R/library/Rcpp/include' -I'$PREFIX/lib/R/library/Rhdf5lib/include' -D_FORTIFY_SOURCE=2 -isystem $PREFIX/include -mmacosx-version-min=10.13 -mmacosx-version-min=10.13 -I$PREFIX/include    -fPIC  -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem $PREFIX/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/r-base-split_1720307552622/work=/usr/local/src/conda/r-base-4.4.1 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix   -c RcppExports.cpp -o RcppExports.o
```

VinzentRisch commented:
Hey @aliciaaevans 
Thank you so much for your help and effort! I hope a solution can be found.

aliciaaevans commented:
@VinzentRisch I'm honestly not sure. I did take a quick look upstream but didn't see anything obvious. You can open an issue with them and link to this PR if you'd like. 

Originally I suspected it was because the overall Bioconda conda_build_config (which inherits conda-forge pinnings) has the 10.13 SDK, but usually overriding it in the recipe folder's conda_build_config.yaml and/or environment variables will do the trick. I am trying a few other things.

jdblischak commented:
> I've tried everything I can think of to force the 10.15 SDK, but it's still getting `-mmacosx-version-min=10.13` from somewhere.

I've experienced similar problems trying to control `-mmacosx-version-min`. Please this Issue for more details: https://github.com/conda-forge/r-base-feedstock/issues/350

As for fixing the problem, the only idea I had was to try saving `CXXFLAGS` with `-D_LIBCPP_DISABLE_AVAILABILITY` directly in `~/.R/Makevars`, but I see you already tried that in https://github.com/bioconda/bioconda-recipes/pull/53137/commits/102610fc023cb20cafd7018d5d672b9a76b64beb

aliciaaevans commented:
Not sure if I found the right env vars or the Makeconf replacement line is what worked. So, backing out changes slowly until it breaks again.

aliciaaevans commented:
@BiocondaBot please fetch artifacts
